core/utils/helpers.py

```diff
@@ -0,0 +1,200 @@
+import os
+import discord
+from collections import defaultdict
+from core.json_io import JsonIO
+from core import json_flusher
+
+
+GLOBAL_KEY = '__global__'
+SENTINEL = object()
+
+
+class JsonDB(JsonIO):
+    """
+    A DB-like helper class to streamline the saving of json files
+
+    Parameters:
+
+    file_path: str
+      The path of the json file you want to create / access
+    create_dirs: bool=False
+      If True, it will create any missing directory leading to
+      the file you want to create
+    autosave: bool=False
+      If True, any change to the "database" will be queued to the
+      flusher and scheduled for a later write
+    default_value: Optional=None
+      Same behaviour as a defaultdict
```

Comment:
Possible rewording here, not everyone knows what a ``defaultdict`` was in its current context.